msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1261,6 +1267,68 @@ public boolean isSharedDevice() {
         return mPublicClientConfiguration.getIsSharedDevice();
     }
 
+    /**
+     * Returns whether the application supports the QR code scanning + PIN protocol.
+     *
+     * @return true if the device supports the QR code scanning + PIN protocol, false otherwise.
+     */
+    @Override
+    public boolean isQRPinAvailable() {
+        final String methodTag = TAG + ":isQRPinAvailable";
+
+        final CommandParameters params = CommandParametersAdapter.createCommandParameters(
+                mPublicClientConfiguration,
+                mPublicClientConfiguration.getOAuth2TokenCache()
+        );
+
+        final List<BaseController> controllers = new MSALControllerFactory(mPublicClientConfiguration)
+                .getAllControllers(mPublicClientConfiguration.getDefaultAuthority());
+
+
+        final IsQrPinAvailableCommand command = new IsQrPinAvailableCommand(
+                params,
+                controllers,
+                new CommandCallback<Boolean, BaseException>() {
+                    @Override
+                    public void onError(BaseException error) {
+                        Logger.error(methodTag, "Unexpected error on isQRPinAvailable", error);
+                    }
+                    @Override
+                    public void onTaskCompleted(Boolean isQrPinAvailable) {
+                        Logger.info(methodTag, "is QR + PIN available? " + isQrPinAvailable );
+                    }
+                    @Override
+                    public void onCancel() {
+                        // Should not be reached.
+                    }
+                },
+                PublicApiId.PCA_IS_QR_PIN_AVAILABLE
+        );
+
+        final ResultFuture<CommandResult> future = CommandDispatcher.submitSilentReturningFuture(command);
+
+        try {
+            final CommandResult commandResult = future.get();
+            switch (commandResult.getStatus()) {
+                case COMPLETED:
+                    Logger.info(methodTag, "is QR + PIN available? " + commandResult.getResult());
+                    return (Boolean) commandResult.getResult();
+                case ERROR:
+                    Logger.error(methodTag, "Unexpected error on isQRPinAvailable", (Throwable) commandResult.getResult());
```

Comment:
This is throwing me off :/ 

We should be using something similar to kotlin's Result class instead. (separate method for result and exception).

Reply:
I'll address this as a separate PR, I would like to keep it simple, and this change will affect other thing no related to QR+PIN.